devtools/gradle/src/main/java/io/quarkus/gradle/dependency/ConditionalDependenciesEnabler.java

```diff
@@ -0,0 +1,193 @@
+        Set<Dependency> dependencies = new HashSet<>();
+        for (ExtensionDependency extensionDependency : extensionDependencies) {
+            dependencies.addAll(extensionDependency.conditionalDependencies);
+        }
+        return dependencies;
+    }
+
+    private void enableConditionalDependency(ModuleVersionIdentifier dependency) {
+        ExtensionDependency extension = featureVariants.get(DependencyUtils.asFeatureName(dependency));
+        if (extension == null) {
+            return;
+        }
+        extension.importConditionalDependency(project, dependency);
+    }
+
+    private ExtensionDependency getExtensionInfoOrNull(ResolvedArtifact artifact, File artifactFile,
+            DependencyHandler dependencies) {
+        ModuleVersionIdentifier artifactId = artifact.getModuleVersion().getId();
+        if (artifactFile.isDirectory()) {
+            Path descriptorPath = artifactFile.toPath().resolve(BootstrapConstants.DESCRIPTOR_PATH);
+            if (Files.exists(descriptorPath)) {
+                return loadExtensionInfo(descriptorPath, dependencies, artifactId);
+            }
+        } else {
+            try (FileSystem artifactFs = ZipUtils.newFileSystem(artifactFile.toPath())) {
+                Path descriptorPath = artifactFs.getPath(BootstrapConstants.DESCRIPTOR_PATH);
+                if (Files.exists(descriptorPath)) {
+                    return loadExtensionInfo(descriptorPath, dependencies, artifactId);
+                }
+            } catch (IOException e) {
+                throw new GradleException("Failed to read " + artifactFile, e);
+            }
+        }
+        return null;
+    }
+
+    private ExtensionDependency loadExtensionInfo(Path descriptorPath, DependencyHandler dependencies,
+            ModuleVersionIdentifier exentionId) {
+        final Properties extensionProperties = new Properties();
+        try (BufferedReader reader = Files.newBufferedReader(descriptorPath)) {
+            extensionProperties.load(reader);
+        } catch (IOException e) {
+            throw new GradleException("Failed to load " + descriptorPath, e);
+        }
+        List<Dependency> conditionalDependencies = new ArrayList<>();
+        if (extensionProperties.containsKey(BootstrapConstants.CONDITIONAL_DEPENDENCIES)) {
+            String conditionalDeps = extensionProperties.get(BootstrapConstants.CONDITIONAL_DEPENDENCIES).toString();
+            for (String conditionalDep : conditionalDeps.split(",")) {
+                conditionalDependencies.add(DependencyUtils.create(dependencies, conditionalDep));
+            }
+        }
+        Dependency constraint = null;
+        if (extensionProperties.containsKey(BootstrapConstants.DEPENDENCY_CONDITION)) {
+            constraint = DependencyUtils.create(dependencies,
+                    extensionProperties.getProperty(BootstrapConstants.DEPENDENCY_CONDITION));
```

Comment:
The condition may include multiple artifacts (artifact keys).

Reply:
oh ok, I will create a list here. And what happen when no condition is declared? right now, the conditional dependency won't be importer, should we?

Reply:
In the Maven impl, no condition is currently treated as being "satisfied".

Reply:
ok, I will update that then.

Reply:
just to make sure, if there is multiple conditions, all condition should be met, or if one is met it's ok?

Reply:
To be more precise, a single condition may include one or more artifact keys. All the artifact keys must be found on the classpath for the condition to be satisfied.